DataStructure/src/main/list/ResizableArray.java

```diff
@@ -35,14 +40,19 @@ public void ensureCapacity(){
             list = list2;
         }
     }
+
     /**
      * Adds an element to the array.
      * @param num
      */
-    public void add(int num){
-        ensureCapacity();
-        list[writeIndex] = num;
-        writeIndex++;
+    public boolean add(int num){
+            ensureCapacity();
+            list[writeIndex] = num;
+            writeIndex++;
+            if (list[writeIndex - 1] == num){
+                return true;
+            }
+            return false;
```

Comment:
writeIndex++ will become negative when you use an array of max size = Integer.MAX_VALUE.

Reply:
There is no need to use if else block for true false.

Reply:
Why do you need to verify the element ?

Reply:
This method will have two output - an exception if the array list size = Integer.MAX_VALUE or true.

Reply:
writeindex negative situation is handled but struggling on fixing the ensureCapacity method which will throw the exception when capacity = Integet.MAX_VALUE.